x-pack/plugins/security_solution/server/lib/detection_engine/routes/signals/set_alert_tags_route.ts

```diff
@@ -49,45 +49,50 @@ export const setAlertTagsRoute = (router: SecuritySolutionPluginRouter) => {
         return siemResponse.error({ statusCode: 404 });
       }
 
-      let queryObject;
-      if (query) {
-        queryObject = {
-          bool: {
-            filter: query,
-          },
-        };
-      }
       const tagsToAdd = uniq(tags.tags_to_add);
       const tagsToRemove = uniq(tags.tags_to_remove);
-      try {
-        const body = await esClient.updateByQuery({
-          index: `${DEFAULT_ALERTS_INDEX}-${spaceId}`,
-          refresh: false,
-          body: {
-            script: {
-              params: { tagsToAdd, tagsToRemove },
-              source: `List newTagsArray = []; 
-              if (ctx._source["kibana.alert.workflow_tags"] != null) {
-                for (tag in ctx._source["kibana.alert.workflow_tags"]) {
-                  if (!params.tagsToRemove.contains(tag)) {
-                    newTagsArray.add(tag);
-                  } 
-                }
-                for (tag in params.tagsToAdd) {
-                  if (!newTagsArray.contains(tag)) {
-                    newTagsArray.add(tag)
-                  }
-                }
-                ctx._source["kibana.alert.workflow_tags"] = newTagsArray;
-              } else {
-                ctx._source["kibana.alert.workflow_tags"] = params.tagsToAdd;
-              }
-              `,
-              lang: 'painless',
+
+      const painlessScript = {
```

Comment:
Curiosity question - is there a particular performance bump we get by using painless script here as opposed to just updating the field with the tag values it should have?

Reply:
It's not performance based necessarily, we just need to access the existing tag array for each alert as to not duplicate any additions. Tied in with the api structure, it's the way we're able to add/remove tags for multiple tags all in one call with the inclusion of that mixed dropdown state you commented on up above.